File: A3/Newton.m
function root = Newton(x0, E, imax, f, fp)
i = 1
fprintf( ' iteration   approximation \n')

for i = 1:n
    root = x0 - (f(x0))/(fp(x0)
    fprintf( ' %6.0f %18.8f \n' , i, root )
    
    if 1-x0/root < E
    end
i = i+1
x0 = root

else
    fprintf( ' failed to converge in %g iterations\n', imax)
      
end

File: A3/newton.py
import math

def main():

    i = 1
    x = 1
    e = 0.000001
    imax = 50
    print("iteration" + "   " + "approximation")

    while (i <= imax):

        root = x - (f(x)/fp(x))
        print(i, end="")
        print(" "*12, end="")
        print('%.6f' % root)

        if ((abs(1 - x/root)) < e):
            return root

        i = i+1
        x = root

    print("failed to converge in" + imax + "iterations")

def f (i):
    ans = (math.cos(i + (2 ** 0.5)) + (i**2)/2 + (2 ** 0.5)*i)
    return ans

def fp (i):
    ans = (-(math.sin(i + 2**0.5)) + i + 2**0.5)
    return ans

if __name__ == '__main__':
	main()
